core/src/main/java/org/elasticsearch/common/logging/LogConfigurator.java

```diff
@@ -135,10 +136,32 @@ private static void configureLoggerLevels(Settings settings) {
         }
     }
 
-
+    /**
+     * Set system properties that can be used in configuration files to specify paths and file patterns for log files. We expose three
+     * properties here:
+     * <ul>
+     * <li>
+     * {@code es.logs.base_path} the base path containing the log files
+     * </li>
+     * <li>
+     * {@code es.logs.cluster_name} the cluster name, used as the prefix of log filenames in the default configuration
+     * </li>
+     * <li>
+     * {@code es.logs.node_name} the node name, can be used as part of log filenames (only exposed if {@link Node#NODE_NAME_SETTING} is
+     * explicitly set)
+     * </li>
+     * </ul>
+     *
+     * @param logsPath the path to the log files
+     * @param settings the settings to extract the cluster and node names
+     */
     @SuppressForbidden(reason = "sets system property for logging configuration")
     private static void setLogConfigurationSystemProperty(final Path logsPath, final Settings settings) {
-        System.setProperty("es.logs", logsPath.resolve(ClusterName.CLUSTER_NAME_SETTING.get(settings).value()).toString());
```

Comment:
When this is backported to the 5.x branch (for inclusion in 5.3), I will keep this property around for backwards compatibility purposes and include a note in the migration docs that this property is being removed in 6.0.0. I investigated the possibility of adding deprecation logging for use of this property, but it is not possible without some horrible reflection-based hacks.